src/Parsers/TablePropertiesQueriesASTs.h

```diff
@@ -85,6 +85,15 @@ using ASTShowCreateDictionaryQuery = ASTQueryWithTableAndOutputImpl<ASTShowCreat
 
 class ASTExistsDatabaseQuery : public ASTQueryWithTableAndOutputImpl<ASTExistsDatabaseQueryIDAndQueryNames>
 {
+public:
+    ASTPtr clone() const override
```

Comment:
The fix is not obvious at all because it's technically not here but in the default implementation of ASTQueryWithTableAndOutputImpl::clone. The default implementation does this:

```
auto res = std::make_shared<ASTQueryWithTableAndOutputImpl<AstIDAndQueryNames>>(*this);
```

This means that clone does not create a ASTExistsDatabaseQuery, but a ASTQueryWithTableAndOutputImpl<ASTExistsDatabaseQueryIDAndQueryNames>, which look the same but they are not as they don't have the same virtual pointer table. Later on, when `formatQueryImpl` is called on the cloned AST, we call the function of the base class (and crash) and not the one of the derived class that we intended.